tests/ui/iter_next_slice.rs

```diff
@@ -0,0 +1,26 @@
+#![warn(clippy::iter_next_slice)]
+
+fn main() {
+    // test code goes here
+    let s = [1, 2, 3];
+    let v = vec![1, 2, 3];
+
+    s.iter().next();
+    // Should be replaced by s.get(0)
+
+    s[2..].iter().next();
+    // Should be replaced by s.get(2)
+
+    v[5..].iter().next();
+    // Should be replaced by v.get(5)
+
+    v.iter().next();
+    // Should be replaced by v.get(0)
+
+    s.into_iter().next();
+    // Should trigger `array_into_iter`/`clippy::into_iter_on_ref` lint
```

Comment:
I agree that having this test once would make sense. But with this test, we cannot run `run-rustfix` on this test anymore. I would prefer to run rustfix for this lint over testing the into_iter case. Maybe add this `into_iter().next()` test to the into_iter_on_ref file, if it doesn't exist there yet.

Reply:
Okay, I have moved this line into `tests/ui/into_iter_on_ref.rs` in the new commit.